src/ra_log.erl

```diff
@@ -993,6 +1046,18 @@ incr_counter(#cfg{counter = undefined}, _Ix, _N) ->
 server_data_dir(Dir, UId) ->
     Me = ra_lib:to_list(UId),
     filename:join(Dir, Me).
+
+maps_with_values(Keys, Map) ->
+    lists:foldr(
+      fun (K, Acc) ->
+              case Map of
+                  #{K := Value} ->
+                      [Value | Acc];
+                  _ ->
+                      Acc
```

Comment:
Is it possible that a requested index has no corresponding entry?

Reply:
`sparse_read` is only used by the `log` effect which in turn is only emitted by state machines which should know the range of available indexes as they control when snapshots can be taken. So no it shouldn't be possible and we can probably remove this clause.

Reply:
We could introduce some code to make the failure more explicit as was the case for the `log` effect previously. It exited with `failed_to_read_index_for_log_effect`